pkg/controller/nodes/task/plugin_config.go

```diff
@@ -8,31 +8,35 @@ import (
 
 	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
 	"github.com/lyft/flytestdlib/logger"
-	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/lyft/flytepropeller/pkg/controller/nodes/task/config"
 	"github.com/lyft/flytepropeller/pkg/controller/nodes/task/k8s"
 )
 
 func WranglePluginsAndGenerateFinalList(ctx context.Context, cfg *config.TaskPluginConfig, pr PluginRegistryIface) ([]core.PluginEntry, error) {
 	allPluginsEnabled := false
-	enabledPlugins := sets.NewString()
+	pluginsConfigMeta := config.PluginsConfigMeta{}
```

Comment:
Should this be:
```suggestion
	pluginsConfigMeta := config.PluginsConfigMeta{
	    AllDefaultForTaskTypes: map[PluginID][]TaskType{}
	}
```
To avoid a nil ref below in case `cfg == nil` ?